apps/ex_marketer/lib/ex_marketer/worker/crawler.ex

```diff
@@ -1,52 +1,57 @@
-defmodule ExMarketer.Crawler.Worker do
+defmodule ExMarketer.Worker.Crawler do
+  use Oban.Worker, queue: :crawler, max_attempts: 2
+
   alias ExMarketer.Keyword
   alias ExMarketer.Crawler.Parse
 
-  def perform(keyword_id, keyword) do
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"keyword_id" => keyword_id, "keyword" => keyword}}) do
     on_start(keyword_id)
 
     try do
       {:ok, response_body} = google_client().get(keyword)
-      result = Parse.perform(response_body)
 
-      on_success(keyword_id, result)
+      on_success(keyword_id, Parse.perform(response_body))
     rescue
       ex ->
-        on_fail(keyword_id, ex)
+        on_fail(keyword_id)
 
-        # Re-raise the exception so the Supervisor could restart the worker
+        # Re-raise the exception so Oban could retry
         raise ex
     end
+
+    :ok
   end
 
   defp google_client, do: Application.get_env(:ex_marketer, :google_client)
 
   defp on_start(keyword_id) do
-    keyword_id
-    |> find_keyword
+    keyword_record = find_keyword(keyword_id)
+
+    keyword_record
     |> Keyword.update!(%{status: Keyword.statues().in_progress})
```

Comment:
Avoid single pipe usage, the following is easier to read.

```
Keyword.update!(keyword_record, %{status: Keyword.statues().in_progress})
```

`|>` is nice but don't overuse it.

Reply:
I think it depends on the first parameter if the first parameter is a function then I think the pipe operator would be nice, but in this case, as it's just a variable so I think we could go without the pipe